lib/Model/IMAPMessage.php

```diff
@@ -619,7 +619,9 @@ private function loadBodyData(Horde_Mime_Part $p, $partNo): string {
 		$p->setContents($data);
 		$data = $p->getContents();
 
-		$data = mb_convert_encoding($data, 'UTF-8', $p->getCharset());
```

Comment:
is  `$p->getCharset()` not reliable?

Reply:
No not really, it uses the 'content-type' header from the email, which can contain invalid charsets / charsets that are not valid for PHP at least.

I would like to combine these two - check for the Horde Charset first, then if that is invalid, use the mb_detect_charset function as it can lead to errors too.